charts/scripts/install-prereqs

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+sa_repo=oci://us-docker.pkg.dev/forgeops-public/charts/secret-agent
+sa_namespace=secret-agent
+sa_version="v1.2.0"
+sa_installed=$(kubectl get crd secretagentconfigurations.secret-agent.secrets.forgerock.io 2>/dev/null)
+if [ -n "$sa_installed" ]; then
+    echo "secret-agent appears to have been previously installed."
+else
+    echo "secret-agent not found.  Installing..."
+    kubectl create namespace $sa_namespace 2>/dev/null || true
+    helm upgrade secret-agent $sa_repo \
+        ${sa_version:+--version=$sa_version} \
+        --namespace $sa_namespace --install --reset-values \
+        \
+        --set tolerations[0]."key"="kubernetes\.io/arch" \
+        --set tolerations[0]."effect"="NoSchedule" \
+        --set tolerations[0]."operator"="Exists"
+fi
+
+cm_repo=https://charts.jetstack.io
+cm_namespace=cert-manager
+cm_version=
+cm_installed=$(kubectl get crd -l app=cert-manager 2>/dev/null)
+if [ -n "$cm_installed" ]; then
+    echo "cert-manager appears to have been previously installed."
+else
+    echo "cert-manager not found.  Installing..."
+    kubectl create namespace $cm_namespace 2>/dev/null || true
+    helm upgrade cert-manager cert-manager --repo $cm_repo \
+        --namespace $cm_namespace --install --reset-values \
+        ${cm_version:+--version=$cm_version} \
+        --set installCRDs=true \
+        --set global.leaderElection.namespace=$cm_namespace \
+        \
+        --set tolerations[0]."key"="kubernetes\.io/arch" \
+        --set tolerations[0]."effect"="NoSchedule" \
+        --set tolerations[0]."operator"="Exists" \
+        --set caininjector.tolerations[0]."key"="kubernetes\.io/arch" \
+        --set caininjector.tolerations[0]."effect"="NoSchedule" \
+        --set caininjector.tolerations[0]."operator"="Exists" \
+        --set startupapicheck.tolerations[0]."key"="kubernetes\.io/arch" \
+        --set startupapicheck.tolerations[0]."effect"="NoSchedule" \
+        --set startupapicheck.tolerations[0]."operator"="Exists" \
+        --set webhook.tolerations[0]."key"="kubernetes\.io/arch" \
+        --set webhook.tolerations[0]."effect"="NoSchedule" \
+        --set webhook.tolerations[0]."operator"="Exists"
+fi
+
+nx_repo=https://kubernetes.github.io/ingress-nginx
+nx_namespace=ingress-nginx
+nx_version=
+nx_installed=$(kubectl get clusterroles -l app.kubernetes.io/name=ingress-nginx 2>/dev/null)
+if [ -n "$nx_installed" ]; then
+    echo "ingress-nginx appears to have been previously installed."
```

Comment:
Could you change this to "NGINX ingress has already been installed."

Reply:
Question - there are going to be some cases where users want to use HAProxy, in which case we don't want to install NGINX (or check for it) but do want to install HAProxy (or at least check for it). Is that doable?

Reply:
A couple of thoughts about this - 

- You could have --nginx and --haproxy options that would check fron the desired ingress controller
- If you didn't feel like scripting adding HAProxy, we could just tell users they have to install HAProxy before continuing

Reply:
Updates made to provide HAProxy option.  Also updated the verbiage.